safedelete/queryset.py

```diff
@@ -107,14 +110,16 @@ def _filter_visibility(self):
                 )
             )
 
+            self._safedelete_filter_applied = True
```

Comment:
I think it's better to use `self._safedelete_filter_applied` in order to make the code more comprehensible. Otherwise knowledge of how `result_cache` works internally is required.